common/primitive-types/src/primitives.rs

```diff
@@ -315,11 +315,23 @@ impl FromStr for Balance {
 
 /// Represents and Ethereum challenge.
 /// This is a one-way encoding of the secp256k1 curve point to an Ethereum address.
-#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
 pub struct EthereumChallenge {
     challenge: [u8; Self::SIZE],
 }
```

Comment:
Why not making this `pub struct EthereumChallenge([u8; Self::SIZE]);` ?

Also the same for the other structs with a single member.

Reply:
good point